src/librustc_mir/borrow_check.rs

```diff
@@ -174,13 +174,17 @@ impl<'c, 'b, 'a: 'b+'c, 'gcx, 'tcx: 'a> DataflowResultsConsumer<'b, 'gcx>
         match stmt.kind {
             StatementKind::Assign(ref lhs, ref rhs) => {
                 self.mutate_lvalue(ContextKind::AssignLhs.new(location),
-                                   (lhs, span), JustWrite, flow_state);
+                                   (lhs, span), Shallow(None), JustWrite, flow_state);
```

Comment:
AST borrowck basically does a `Deep` write here and we might want to stay in sync with it. Also, if you leave this as `Shallow` without erasing borrows in `dataflow::impls::borrows` you'll have some *very* confusing diagnostics.